java/src/main/java/ai/rapids/cudf/Scalar.java

```diff
@@ -541,6 +565,11 @@ public boolean equals(Object o) {
       return getLong() == other.getLong();
     case STRING:
       return Arrays.equals(getUTF8(), other.getUTF8());
+    case LIST:
+      try (ColumnView viewMe = getListAsColumnView();
+           ColumnView viewO = other.getListAsColumnView()) {
+        return viewMe.equals(viewO);
```

Comment:
Related to the hashCode/equals comment above, this should do the legwork of comparing the columns, otherwise this is going to throw in many cases which is unhelpful.

Reply:
Yes, you are right. The issue https://github.com/rapidsai/cudf/issues/8090 is filed for this.